File: task/src/task_status.rs
use std::str::FromStr;

#[derive(PartialEq, Clone, Debug)]
pub enum TaskStatus {
    Pending,
    Running,
    Retrying,
    Success,
    Failure,
    Skipped,
}

impl TaskStatus {
    // Convert a TaskStatus to a &str
    pub fn as_str(&self) -> &'static str {
        match *self {
            TaskStatus::Pending => "Pending",
            TaskStatus::Running => "Running",
            TaskStatus::Retrying => "Retrying",
            TaskStatus::Success => "Success",
            TaskStatus::Failure => "Failure",
            TaskStatus::Skipped => "Skipped",
        }
    }

    pub fn as_u8(&self) -> u8 {
        match *self {
            TaskStatus::Pending => 0,
            TaskStatus::Running => 1,
            TaskStatus::Retrying => 2,
            TaskStatus::Success => 3,
            TaskStatus::Failure => 4,
            TaskStatus::Skipped => 5,
        }
    }
}

impl FromStr for TaskStatus {
    type Err = ();

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s {
            "Pending" => Ok(TaskStatus::Pending),
            "Running" => Ok(TaskStatus::Running),
            "Retrying" => Ok(TaskStatus::Retrying),
            "Success" => Ok(TaskStatus::Success),
            "Failure" => Ok(TaskStatus::Failure),
            "Skipped" => Ok(TaskStatus::Skipped),
            _ => Err(()),
        }
    }
}

File: thepipelinetool_ui/lib/classes/task_status.dart
// const int maxNumber = 400;

// ignore_for_file: constant_identifier_names

enum TaskStatus {
  Pending,
  Running,
  Retrying,
  Success,
  Failure,
  Skipped,
  None;

  // Static method to get enum from int
  static TaskStatus fromInt(int index) {
    if (index < 0 || index >= TaskStatus.values.length) {
      throw ArgumentError('Invalid index for MyEnum: $index');
    }
    return TaskStatus.values[index];
  }

  @override
  String toString() {
    switch (this) {
      case TaskStatus.Pending:
        return "Pending";
      case TaskStatus.Running:
        return "Running";
      case TaskStatus.Retrying:
        return "Retrying";
      case TaskStatus.Success:
        return "Success";
      case TaskStatus.Failure:
        return "Failure";
      case TaskStatus.Skipped:
        return "Skipped";
      case TaskStatus.None:
        return '';
    }
  }
}

extension IntoTaskStatus on int {
  TaskStatus toTaskStatus() => TaskStatus.fromInt(this);
}
